application/Espo/Modules/Postgres/ORM/QueryComposer/PostgresqlQueryComposer.php

```diff
@@ -0,0 +1,808 @@
+        }
+
+        /*if (!empty($indexKeyList)) { //HERE - index hinting not supported
+            foreach ($indexKeyList as $index) {
+                $sql .= " USE INDEX (" . $this->quoteIdentifier($this->sanitizeIndexName($index)) . ")";
+            }
+        }*/
+
+        if (!empty($joins)) {
+            $sql .= " {$joins}";
+        }
+
+        if ($where !== null && $where !== '') {
+            $sql .= " WHERE {$where}";
+        }
+
+        if (!empty($groupBy)) {
+            $sql .= " GROUP BY {$groupBy}";
+        }
+
+        if ($having !== null && $having !== '') {
+            $sql .= " HAVING {$having}";
+        }
+
+        if (!empty($order)) {
+            $sql .= " ORDER BY {$order}";
+        }
+
+        if (is_null($offset) && !is_null($limit)) {
+            $offset = 0;
+        }
+
+        $sql = $this->limit($sql, $offset, $limit);
+
+        if ($tailPart) {
+            $sql .= " " . $tailPart;
+        }
+
+        return $sql;
+    }
+
+    protected function getFunctionPart(
+        string $function,
+        string $part,
+        array  $params,
+        string $entityType,
+        bool   $distinct,
+        array  $argumentPartList = []
+    ): string
+    {
+        if ($function === 'POSITION_IN_LIST') { //HERE - to be rewritten, postgresql does not support FIELD
+            $sql = 'CASE';
+            $column = $this->quoteColumn($params[0]);
+
+            foreach (array_slice($argumentPartList, 1) as $i => $argumentPart) {
+                $argumentPart = $this->quote($argumentPart);
+
+                $sql .= " WHEN $column = $argumentPart THEN $i";
+            }
+
+            $sql .= 'ELSE 0 END';
+            return $sql;
+        }
+
+        return parent::getFunctionPart($function, $part, $params, $entityType, $distinct, $argumentPartList);
+    }
+
+    protected function getOrderExpressionPart(
+        Entity $entity,
+               $orderBy = null,
+               $order = null,
+        ?array &$params = null,
+        bool   $noCustom = false): ?string
+    {
+        if (is_null($orderBy)) {
+            return null;
+        }
+
+        if (is_array($orderBy)) {
+            $arr = [];
+
+            foreach ($orderBy as $item) {
+                if (is_array($item)) {
+                    $orderByInternal = $item[0];
+                    $orderInternal = null;
+
+                    if (!empty($item[1])) {
+                        $orderInternal = $item[1];
+                    }
+
+                    $arr[] = $this->getOrderExpressionPart(
+                        $entity,
+                        $orderByInternal,
+                        $orderInternal,
+                        $params,
+                        $noCustom
+                    );
+                }
+            }
+
+            return implode(", ", $arr);
+        }
+
+        if (str_starts_with($orderBy, 'LIST:')) { // HERE - same thing as above, no field function
```

Comment:
I refactored this so that `POSITION_IN_LIST` complex expression function is used instead. `LIST:` is a legacy.

https://github.com/espocrm/espocrm/commit/d82c297b797b5a38d98749f202044403ed1b74f6